frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,42 @@
+import { useCallback } from 'react';
+import { createPortal } from 'react-dom';
+
+import { IconButton } from '~/libs/components/components.js';
+import { useState } from '~/libs/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  children?: React.ReactNode;
+};
+
+const Modal: React.FC<Properties> = ({ children }: Properties) => {
+  const [isOpen, setIsOpen] = useState(true);
+
+  const onClose = useCallback(() => setIsOpen(false), [setIsOpen]);
+
+  return (
+    <>
+      {isOpen &&
+        createPortal(
+          <>
+            <dialog className={styles.modal} id="modal">
+              <div className={styles.modalInner}>
+                <IconButton
+                  icon="cross"
+                  label="Close modal"
+                  className={styles.modalPopUpClose as string}
+                  onClick={onClose}
+                />
+                {children}
+              </div>
+              <div className={styles.backdrop} />
```

Comment:
```suggestion
```
https://developer.mozilla.org/en-US/docs/Web/CSS/::backdrop

Reply:
dialog element has by default set value background-color: canvas and it overwrites my styles used with backdrop pseudo-element. So should I set backdrop as backdround styles for modal class?

Reply:
```
.modal {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  border: none;
  background: color(srgb 0 0 0 / 60%);
}
```

Reply:
you need to use `::backdrop` as the modal background